kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/WatchConnectionManager.java

```diff
@@ -331,7 +331,7 @@ public void waitUntilReady() {
   @Override
   public void close() {
     logger.debug("Force closing the watch {}", this);
-    closeEvent(null);
+    closeEvent();
```

Comment:
I don't think that this addresses the issue because very often what triggers the call to `close` here is an exception, which is still not propagated in that case.

Reply:
I really don't see how an Exception will trigger this close since this method will most probably be called by try-with-resources Closable.

In the event of an Exception triggering this close, the client would also be aware of the Exception via the catch block.

Do you have any code which shows this "because very often what triggers the call to close here is an exception" so I can understand better?